tika-core/src/main/java/org/apache/tika/config/TikaActivator.java

```diff
@@ -51,14 +53,17 @@ public void start(final BundleContext context) throws Exception {
 
         detectorTracker = new ServiceTracker(context, Detector.class.getName(), this);
         parserTracker = new ServiceTracker(context, Parser.class.getName(), this);
+        zipDetectorTracker = new ServiceTracker(context, "org.apache.tika.detect.zip.ZipContainerDetector", this);
```

Comment:
Do we need to do this for EncodingDetector as well to ensure that all encoding detectors are dynamically loaded?  Obv, separate pull request...but I'm curious if that is failing in OSGi.